Tests/TestAddressBook.cpp

```diff
@@ -0,0 +1,745 @@
+/* Tests the AddressBook for functionality and integrity. */
+
+#define CATCH_CONFIG_MAIN
+
+#include "catch.hpp"
+
+#include "OSFixes.hpp"
+#include "AddressBook.hpp"
+#include "AddressBook_Defs.hpp"
+#include "AddressBook_Task.hpp"
+
+#include <stdio.h>
+#include <cstddef>
+#include <string>
+#include <iostream>
+#include <iomanip>
+#include <vector>
+#include <numeric>
+#include <iterator>
+#include <algorithm>
+#include <map>
+
+
+TEST_CASE("AddressBook Small Tests", "[Simple]")
+{
+    // Create the AddressBook instance
+    AddressBook::AddressBook AddrBook
+        = AddressBook::AddressBook(std::string("AddressBookMain"));
+    AddressBook::TaskData_t taskData;
+    
+    //Create a Supervisor devicetype record and add some messagetypes
+    AddressBook::DevTypeRecord_t SuperDTR("Super");
+    SuperDTR.InMsgs.push_back(1);
+    SuperDTR.OuMsgs.push_back(2);
+    
+    //Create a Cell devicetype record and add some messagetytpes
+    AddressBook::DevTypeRecord_t CellDTR("Cell");
+    CellDTR.InMsgs.push_back(0);
+    CellDTR.InMsgs.push_back(2);
+    CellDTR.OuMsgs.push_back(0);
+    CellDTR.OuMsgs.push_back(1);
+    
+    //Create a Fixed node devicetype record and add some messagetytpes
+    AddressBook::DevTypeRecord_t FixedDTR("Fixed");
+    FixedDTR.InMsgs.push_back(2);
+    FixedDTR.OuMsgs.push_back(0);
+    FixedDTR.OuMsgs.push_back(1);
+    
+    
+    
+    //==========================================================================
+    // Add a task (and check that it added correctly)
+    //==========================================================================
+    std::string t1Name = "Task1";
+    AddressBook::TaskData_t T1Data;
+    
+    T1Data.DeviceCount = 5;
+    T1Data.ExternalCount = 2;
+    T1Data.Path = "/local/extra/Orchestrator/application_staging/xml";
```

Comment:
I realise this may well be a dummy, but if not, I recommend dropping static test resources into `Tests/StaticResources`.

Reply:
Yep, it is a complete dummy - just a string that looks vaguely reasonable